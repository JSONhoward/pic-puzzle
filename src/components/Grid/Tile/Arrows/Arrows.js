import React from 'react'
import styled from 'styled-components'
import { useSetRecoilState, useResetRecoilState } from 'recoil'
import { FaAngleUp, FaAngleDown, FaAngleLeft, FaAngleRight } from 'react-icons/fa'
import { TileContent, ShowArrows } from '../../../../Store'

export const StyledArrows = styled.div`
position: absolute;
height: 100%;
width: 100%;
z-index: 1;
`

const ArrowIcon = styled.div`
position: absolute;
top: ${props => props.direction === 'left' || props.direction === 'right' ? '50%' : 'none'};
left: ${props => props.direction === 'left' ? '5px' : props.direction === 'up' || props.direction === 'down' ? '50%' : 'none'};
right: ${props => props.direction === 'right' ? '5px' : 'none'};
bottom: ${props => props.direction === 'down' ? '5px' : 'none'};
transform: ${props => props.direction === 'left' || props.direction === 'right' ? 'translateY(-50%)' : 'translateX(-50%)'};
cursor: pointer;

&:hover {
    color: yellow;
}
`

const Arrows = ({ column, row, tileNum }) => {
    const setTileContent = useSetRecoilState(TileContent)
    const resetArrow = useResetRecoilState(ShowArrows)

    const moveTile = (e, num) => {
        let arrow = e.currentTarget.getAttribute('direction')
        switch (num) {
            case 1:
                if (arrow === 'right') {
                    setTileContent(prevContent => {
                        if (prevContent.tile1 !== null && prevContent.tile2 === null) {
                            let state = {
                                ...prevContent,
                                tile1: null,
                                tile2: prevContent.tile1
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent(prevContent => {
                        if (prevContent.tile1 !== null && prevContent.tile4 === null) {
                            let state = {
                                ...prevContent,
                                tile1: null,
                                tile4: prevContent.tile1
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 2:
                if (arrow === 'right') {
                    setTileContent(prevContent => {
                        if (prevContent.tile2 !== null && prevContent.tile3 === null) {
                            let state = {
                                ...prevContent,
                                tile2: null,
                                tile3: prevContent.tile2
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent(prevContent => {
                        if (prevContent.tile2 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile2: null,
                                tile5: prevContent.tile2
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'left') {
                    setTileContent(prevContent => {
                        if (prevContent.tile2 !== null && prevContent.tile1 === null) {
                            let state = {
                                ...prevContent,
                                tile2: null,
                                tile1: prevContent.tile2
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 3:
                if (arrow === 'left') {
                    setTileContent(prevContent => {
                        if (prevContent.tile3 !== null && prevContent.tile2 === null) {
                            let state = {
                                ...prevContent,
                                tile3: null,
                                tile2: prevContent.tile3
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent(prevContent => {
                        if (prevContent.tile3 !== null && prevContent.tile6 === null) {
                            let state = {
                                ...prevContent,
                                tile3: null,
                                tile6: prevContent.tile3
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 4:
                if (arrow === 'up') {
                    setTileContent(prevContent => {
                        if (prevContent.tile4 !== null && prevContent.tile1 === null) {
                            let state = {
                                ...prevContent,
                                tile4: null,
                                tile1: prevContent.tile4
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent(prevContent => {
                        if (prevContent.tile4 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile4: null,
                                tile5: prevContent.tile4
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent(prevContent => {
                        if (prevContent.tile4 !== null && prevContent.tile7 === null) {
                            let state = {
                                ...prevContent,
                                tile4: null,
                                tile7: prevContent.tile4
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 5:
                if (arrow === 'up') {
                    setTileContent(prevContent => {
                        if (prevContent.tile5 !== null && prevContent.tile2 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile2: prevContent.tile5
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent(prevContent => {
                        if (prevContent.tile5 !== null && prevContent.tile6 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile6: prevContent.tile5
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent(prevContent => {
                        if (prevContent.tile5 !== null && prevContent.tile8 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile8: prevContent.tile5
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'left') {
                    setTileContent(prevContent => {
                        if (prevContent.tile5 !== null && prevContent.tile4 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile4: prevContent.tile5
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 6:
                if (arrow === 'up') {
                    setTileContent(prevContent => {
                        if (prevContent.tile6 !== null && prevContent.tile3 === null) {
                            let state = {
                                ...prevContent,
                                tile6: null,
                                tile3: prevContent.tile6
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent(prevContent => {
                        if (prevContent.tile6 !== null && prevContent.tile9 === null) {
                            let state = {
                                ...prevContent,
                                tile6: null,
                                tile9: prevContent.tile6
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'left') {
                    setTileContent(prevContent => {
                        if (prevContent.tile6 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile6: null,
                                tile5: prevContent.tile6
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 7:
                if (arrow === 'up') {
                    setTileContent(prevContent => {
                        if (prevContent.tile7 !== null && prevContent.tile4 === null) {
                            let state = {
                                ...prevContent,
                                tile7: null,
                                tile4: prevContent.tile7
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent(prevContent => {
                        if (prevContent.tile7 !== null && prevContent.tile8 === null) {
                            let state = {
                                ...prevContent,
                                tile7: null,
                                tile8: prevContent.tile7
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 8:
                if (arrow === 'left') {
                    setTileContent(prevContent => {
                        if (prevContent.tile8 !== null && prevContent.tile7 === null) {
                            let state = {
                                ...prevContent,
                                tile8: null,
                                tile7: prevContent.tile8
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'up') {
                    setTileContent(prevContent => {
                        if (prevContent.tile8 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile8: null,
                                tile5: prevContent.tile8
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent(prevContent => {
                        if (prevContent.tile8 !== null && prevContent.tile9 === null) {
                            let state = {
                                ...prevContent,
                                tile8: null,
                                tile9: prevContent.tile8
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 9:
                if (arrow === 'left') {
                    setTileContent(prevContent => {
                        if (prevContent.tile9 !== null && prevContent.tile8 === null) {
                            let state = {
                                ...prevContent,
                                tile9: null,
                                tile8: prevContent.tile9
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'up') {
                    setTileContent(prevContent => {
                        if (prevContent.tile9!== null && prevContent.tile6 === null) {
                            let state = {
                                ...prevContent,
                                tile9: null,
                                tile6: prevContent.tile9
                            }
                            resetArrow()
                            return state
                        }
                        return prevContent
                    })
                }
                break
            default:
                break
        }
    }

    return (
        <StyledArrows>
            {column !== 1 && <ArrowIcon onClick={e => moveTile(e, tileNum)} direction={'left'}><FaAngleLeft size={'3rem'} /></ArrowIcon>}
            {row !== 1 && <ArrowIcon onClick={e => moveTile(e, tileNum)} direction={'up'}><FaAngleUp size={'3rem'} /></ArrowIcon>}
            {column !== 3 && <ArrowIcon onClick={e => moveTile(e, tileNum)} direction={'right'}><FaAngleRight size={'3rem'} /></ArrowIcon>}
            {row !== 3 && <ArrowIcon onClick={e => moveTile(e, tileNum)} direction={'down'}><FaAngleDown size={'3rem'} /></ArrowIcon>}
        </StyledArrows>
    )
}

export default Arrows
