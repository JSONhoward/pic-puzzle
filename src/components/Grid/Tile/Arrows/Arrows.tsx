import * as React from 'react'
import styled from 'styled-components'
import { useSetRecoilState, useResetRecoilState } from 'recoil'
import { FaAngleUp, FaAngleDown, FaAngleLeft, FaAngleRight } from 'react-icons/fa'

import { TileContent, ShowArrows, TileMoves } from '../../../../Store/index'

export const StyledArrows = styled.div`
position: absolute;
height: 100%;
width: 100%;
z-index: 1;
`

const ArrowIcon = styled('div')<{direction: String}>`
position: absolute;
top: ${props => props.direction === 'left' || props.direction === 'right' ? '50%' : 'none'};
left: ${props => props.direction === 'left' ? '5px' : props.direction === 'up' || props.direction === 'down' ? '50%' : 'none'};
right: ${props => props.direction === 'right' ? '5px' : 'none'};
bottom: ${props => props.direction === 'down' ? '5px' : 'none'};
transform: ${props => props.direction === 'left' || props.direction === 'right' ? 'translateY(-50%)' : 'translateX(-50%)'};
cursor: pointer;
color: rgb(30,30,30);

&:hover {
    color: red;
}
`

interface ArrowsProps {
    column: Number,
    row: Number,
    tileNum: Number
}

const Arrows: React.FC<ArrowsProps> = ({ column, row, tileNum }) => {
    const setTileContent = useSetRecoilState(TileContent)
    const resetArrow = useResetRecoilState(ShowArrows)
    const incrementMoves = useSetRecoilState(TileMoves)

    const moveTile = (e: React.MouseEvent<HTMLDivElement>): void => {
        e.preventDefault()
        const element = e.currentTarget as HTMLElement
        let arrow = element.getAttribute('direction')
        switch (tileNum) {
            case 1:
                if (arrow === 'right') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile1 !== null && prevContent.tile2 === null) {
                            let state = {
                                ...prevContent,
                                tile1: null,
                                tile2: prevContent.tile1
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile1 !== null && prevContent.tile4 === null) {
                            let state = {
                                ...prevContent,
                                tile1: null,
                                tile4: prevContent.tile1
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 2:
                if (arrow === 'right') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile2 !== null && prevContent.tile3 === null) {
                            let state = {
                                ...prevContent,
                                tile2: null,
                                tile3: prevContent.tile2
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile2 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile2: null,
                                tile5: prevContent.tile2
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'left') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile2 !== null && prevContent.tile1 === null) {
                            let state = {
                                ...prevContent,
                                tile2: null,
                                tile1: prevContent.tile2
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 3:
                if (arrow === 'left') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile3 !== null && prevContent.tile2 === null) {
                            let state = {
                                ...prevContent,
                                tile3: null,
                                tile2: prevContent.tile3
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile3 !== null && prevContent.tile6 === null) {
                            let state = {
                                ...prevContent,
                                tile3: null,
                                tile6: prevContent.tile3
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 4:
                if (arrow === 'up') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile4 !== null && prevContent.tile1 === null) {
                            let state = {
                                ...prevContent,
                                tile4: null,
                                tile1: prevContent.tile4
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile4 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile4: null,
                                tile5: prevContent.tile4
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile4 !== null && prevContent.tile7 === null) {
                            let state = {
                                ...prevContent,
                                tile4: null,
                                tile7: prevContent.tile4
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 5:
                if (arrow === 'up') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile5 !== null && prevContent.tile2 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile2: prevContent.tile5
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile5 !== null && prevContent.tile6 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile6: prevContent.tile5
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile5 !== null && prevContent.tile8 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile8: prevContent.tile5
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'left') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile5 !== null && prevContent.tile4 === null) {
                            let state = {
                                ...prevContent,
                                tile5: null,
                                tile4: prevContent.tile5
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 6:
                if (arrow === 'up') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile6 !== null && prevContent.tile3 === null) {
                            let state = {
                                ...prevContent,
                                tile6: null,
                                tile3: prevContent.tile6
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'down') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile6 !== null && prevContent.tile9 === null) {
                            let state = {
                                ...prevContent,
                                tile6: null,
                                tile9: prevContent.tile6
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'left') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile6 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile6: null,
                                tile5: prevContent.tile6
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 7:
                if (arrow === 'up') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile7 !== null && prevContent.tile4 === null) {
                            let state = {
                                ...prevContent,
                                tile7: null,
                                tile4: prevContent.tile7
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile7 !== null && prevContent.tile8 === null) {
                            let state = {
                                ...prevContent,
                                tile7: null,
                                tile8: prevContent.tile7
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 8:
                if (arrow === 'left') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile8 !== null && prevContent.tile7 === null) {
                            let state = {
                                ...prevContent,
                                tile8: null,
                                tile7: prevContent.tile8
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'up') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile8 !== null && prevContent.tile5 === null) {
                            let state = {
                                ...prevContent,
                                tile8: null,
                                tile5: prevContent.tile8
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'right') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile8 !== null && prevContent.tile9 === null) {
                            let state = {
                                ...prevContent,
                                tile8: null,
                                tile9: prevContent.tile8
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            case 9:
                if (arrow === 'left') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile9 !== null && prevContent.tile8 === null) {
                            let state = {
                                ...prevContent,
                                tile9: null,
                                tile8: prevContent.tile9
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                } else if (arrow === 'up') {
                    setTileContent((prevContent: any) => {
                        if (prevContent.tile9!== null && prevContent.tile6 === null) {
                            let state = {
                                ...prevContent,
                                tile9: null,
                                tile6: prevContent.tile9
                            }
                            resetArrow()
                            incrementMoves((prevCount: number) => prevCount + 1)
                            return state
                        }
                        return prevContent
                    })
                }
                break
            default:
                break
        }
    }

    return (
        <StyledArrows>
            {column !== 1 && <ArrowIcon onClick={(event: React.MouseEvent<HTMLDivElement>): void => moveTile(event)} direction={'left'}><FaAngleLeft size={'3rem'} /></ArrowIcon>}
            {row !== 1 && <ArrowIcon onClick={(event: React.MouseEvent<HTMLDivElement>): void => moveTile(event)} direction={'up'}><FaAngleUp size={'3rem'} /></ArrowIcon>}
            {column !== 3 && <ArrowIcon onClick={(event: React.MouseEvent<HTMLDivElement>): void => moveTile(event)} direction={'right'}><FaAngleRight size={'3rem'} /></ArrowIcon>}
            {row !== 3 && <ArrowIcon onClick={(event: React.MouseEvent<HTMLDivElement>): void => moveTile(event)} direction={'down'}><FaAngleDown size={'3rem'} /></ArrowIcon>}
        </StyledArrows>
    )
}

export default Arrows
