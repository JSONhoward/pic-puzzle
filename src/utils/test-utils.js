import React from 'react'
import renderer from 'react-test-renderer'
import { RecoilRoot } from 'recoil'

export const renderWrapper = (elements) => {
    return renderer.create(<RecoilRoot>{elements}</RecoilRoot>)
}