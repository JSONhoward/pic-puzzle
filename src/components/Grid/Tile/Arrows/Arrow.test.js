import React from 'react'
import Arrows from '../Arrows/Arrows'
import {renderWrapper} from '../../../../utils/test-utils'

describe('Arrows', () => {
    test('snapshot renders', () => {
        const component = renderWrapper(<Arrows />)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})