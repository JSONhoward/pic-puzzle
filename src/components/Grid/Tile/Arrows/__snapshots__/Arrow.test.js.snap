// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrows snapshot renders 1`] = `
<div
  className="sc-AxjAm hAemoA"
>
  <div
    className="sc-AxirZ cQftid"
    direction="left"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="3rem"
      size="3rem"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 256 512"
      width="3rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
      />
    </svg>
  </div>
  <div
    className="sc-AxirZ kMBLbH"
    direction="up"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="3rem"
      size="3rem"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="3rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
      />
    </svg>
  </div>
  <div
    className="sc-AxirZ EUpmt"
    direction="right"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="3rem"
      size="3rem"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 256 512"
      width="3rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      />
    </svg>
  </div>
  <div
    className="sc-AxirZ dNYJkA"
    direction="down"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="3rem"
      size="3rem"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="3rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
      />
    </svg>
  </div>
</div>
`;
